File: spring-boot-3-demo-app/src/main/java/io/github/mfvanek/spring/boot3/test/controllers/RedirectController.java
package io.github.mfvanek.spring.boot3.test.controllers;

import org.springframework.http.HttpHeaders;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import java.net.URI;
import java.net.URISyntaxException;

@RestController
public class RedirectController {

    // http://localhost:8080/redirect
    @GetMapping(path = "/redirect")
    public ResponseEntity<Object> redirectToGoogle() throws URISyntaxException {
        final URI google = new URI("https://www.google.com");
        final HttpHeaders httpHeaders = new HttpHeaders();
        httpHeaders.setLocation(google);
        return new ResponseEntity<>(httpHeaders, HttpStatus.SEE_OTHER);
    }
}

File: spring-boot-3-demo-app-kotlin/src/main/kotlin/io/github/mfvanek/spring/boot3/kotlin/test/controllers/RedirectController.kt
/*
 * Copyright (c) 2020-2025. Ivan Vakhrushev and others.
 * https://github.com/mfvanek/spring-boot-open-telemetry-demo
 *
 * Licensed under the Apache License 2.0
 */

package io.github.mfvanek.spring.boot3.kotlin.test.controllers

import java.net.URI
import org.springframework.http.HttpHeaders
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class RedirectController {
    // http://localhost:8080/redirect
    @GetMapping(path = ["/redirect"])
    fun redirectToGoogle(): ResponseEntity<Any> {
        val google = URI("https://www.google.com")
        val httpHeaders = HttpHeaders()
        httpHeaders.location = google
        return ResponseEntity(httpHeaders, HttpStatus.SEE_OTHER)
    }
}
